src/rustup-cli/setup_mode.rs

```diff
@@ -44,7 +43,21 @@ pub fn main() -> Result<()> {
     let default_host = matches.value_of("default-host").map(|s| s.to_owned()).unwrap_or_else(|| {
         TargetTriple::from_host_or_build().to_string()
     });
-    let default_toolchain = matches.value_of("default-toolchain").unwrap_or("stable");
+    let default_toolchain = match matches.value_of("default-toolchain") {
+        Some(default_toolchain) => {
+            if !["stable", "beta", "nightly"].contains(&default_toolchain) {
+                println!(
+                    "Warning: You've specified a custom default toolchain. \
+                    Custom toolchains are described with a specific syntax. \
+                    If Rust fails to install, you may want to consider installing one of the \
+                    named release channels instead: stable, beta, or nightly."
+                );
+            }
+
+            default_toolchain
+        }
+        None => "stable",
+    };
```

Comment:
If you don't mind, let's take this warning out. I'm afraid it will look a bit too threatening. For now let's just not do any new error handling, even though if the toolchain fails to install the failure mode is pretty messy.

Instead, I'd prefer to make the way self_update responds to a failed toolchain install more robust, so that, *if* the toolchain install fails, the installer prints a message explaining what happened and how to proceed. But that's a pretty big patch on its own and I think can be done later.